### ISSUE TITLE
Add `eslint-plugin-tsdoc`

### ISSUE THREAD

**Issue by alecmev** [davidjbradshaw/eslint-config-auto#56]
I think it fits in, right? https://github.com/microsoft/tsdoc/tree/main/eslint-plugin

**Comment by davidjbradshaw** [comment]
Yeah I think it would be reasonable to detect TSDoc and load the limiting rules in as required. I guess we should also do the same with JSDoc

Would you like to make a PR for both?

**Comment by alecmev** [comment]
> detect TSDoc

Hmm, I'm not sure how to go about that. We stick to the standard for better hints, but we don't generate any documentation, so there's no `@microsoft/tsdoc` in our lockfile. Is there a downside to enabling it for all TypeScript projects? It isn't obtrusive, just in case, I don't remember it complaining about something it shouldn't complain about.

> I guess we should also do the same with JSDoc

Sure! Same question about the detection though.

**Comment by davidjbradshaw** [comment]
I think the downside is not everyone uses TSdoc or JSdoc. So we don’t want to force this onto them.
